File: app/src/main/java/com/jkhub/webhook/MainActivity.java
package com.jkhub.webhook;

import android.app.Activity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        EditText webhookUrl = findViewById(R.id.webhookUrl);
        EditText signalType = findViewById(R.id.signalType);
        EditText pair = findViewById(R.id.pair);
        EditText entry = findViewById(R.id.entry);
        EditText sl = findViewById(R.id.sl);
        EditText tp1 = findViewById(R.id.tp1);
        EditText tp2 = findViewById(R.id.tp2);

        Button sendButton = findViewById(R.id.sendButton);
        TextView status = findViewById(R.id.status);

        sendButton.setOnClickListener(v -> {

            String url = webhookUrl.getText().toString().trim();
            String type = signalType.getText().toString().trim();
            String symbol = pair.getText().toString().trim();
            String entryPrice = entry.getText().toString().trim();

            if (url.isEmpty() || symbol.isEmpty() || entryPrice.isEmpty()) {
                status.setText("Please fill required fields");
                return;
            }

            status.setText("Signal ready: " + type + " " + symbol);
        });
    }
}

File: app/src/main/java/com/jkhub/webhook/MainActivity.kt
package com.jkhub.webhook

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        val webhookUrl = findViewById<EditText>(R.id.webhookUrl)
        val signalType = findViewById<EditText>(R.id.signalType)
        val pair = findViewById<EditText>(R.id.pair)
        val entry = findViewById<EditText>(R.id.entry)
        val sl = findViewById<EditText>(R.id.sl)
        val tp1 = findViewById<EditText>(R.id.tp1)
        val tp2 = findViewById<EditText>(R.id.tp2)
        val sendButton = findViewById<Button>(R.id.sendButton)
        val status = findViewById<TextView>(R.id.status)

        sendButton.setOnClickListener {

            val url = webhookUrl.text.toString().trim()
            val type = signalType.text.toString().trim()
            val symbol = pair.text.toString().trim()
            val entryPrice = entry.text.toString().trim()

            if (url.isEmpty() || symbol.isEmpty() || entryPrice.isEmpty()) {
                status.text = "Please fill required fields"
                return@setOnClickListener
            }

            status.text = "Signal ready: $type $symbol"
        }
    }
}                return@setOnClickListener
            }

            status.text = "Signal ready: $type $symbol"
        }
    }
}
